cdap-docs/admin-manual/source/operations/tx-maintenance.rst

```diff
@@ -10,56 +10,98 @@ Transaction Service Maintenance
+ 
+#. Run a flush and a major compaction on all CDAP transactional tables.
+ 
+#. Wait for the major compaction to complete.
+ 
+#. Find the minimum prune time across all queues by running the tool
+   ``SimpleHBaseQueueDebugger`` (note that authorization is disabled when running the tool
+   so that CDAP can read all users' queue tables)::
+ 
+ 
+     $ /opt/cdap/master/bin/cdap run co.cask.cdap.data.tools.SimpleHBaseQueueDebugger
+ 
+     Results for queue queue:///ns1/WordCount/WordCounter/counter/queue: min tx timestamp: 1440198510309
+     Results for queue queue:///ns1/WordCount/WordCounter/splitter/wordArrayOut: min tx timestamp: 1440198510280
+     Results for queue queue:///ns2/WordCount/WordCounter/counter/queue: min tx timestamp: n/a
+     Results for queue queue:///ns2/WordCount/WordCounter/splitter/wordArrayOut: min tx timestamp: n/a
+     Results for queue queue:///default/WordCount/WordCounter/counter/queue: min tx timestamp: 1440194184568
+     Results for queue queue:///default/WordCount/WordCounter/splitter/wordArrayOut: min tx timestamp: 1440194184476
+     Results for queue queue:///default/WordCount/WordCounter/splitter/wordOut: min tx timestamp: 1440194184476
+     Total results for all queues: min tx timestamp: 1440194184476
+ 
+   Pick the timestamp from the line beginning ``Total results for all queues``. In this case, ``1440194184476``.
+ 
+#. Get the minimum time from the above two steps, to obtain the ``pruneTime``. In this case, ``1440194184476``.
+ 
+#. If the CDAP tables are replicated to other clusters, see the section below
+   (:ref:`tx-maintenance-pruning-replicated`) to obtain the ``pruneTime`` for the slave
+   clusters.
+ 
+#. The final ``pruneTime`` is the minimum ``pruneTime`` across all replicated clusters (if
+   any); let this be time ``t``.
+
+Now, the invalid transaction list can be safely pruned until ``(t - 1 day)`` using a call
+to :ref:`truncate invalid transactions before a specific time
+<http-restful-api-transactions-truncate>`.
+
+The current length of the invalid transaction list can be obtained using a call to
+:ref:`retrieve the number of invalid transactions <http-restful-api-transactions-number>`.
+
+.. _tx-maintenance-pruning-replicated:
+
+Pruning Invalid Transactions in a Replicated Cluster
+----------------------------------------------------
+1. Copy over the latest transaction snapshots from the master cluster to the slave cluster.
+
+#. Wait for three to four minutes for the latest transaction state to be reloaded from the
+   snapshot.
+
+#. Run steps 1 to 5 from the above section (:ref:`tx-maintenance-pruning-invalid-transactions`)
+   on the slave cluster to find the ``pruneTime`` for that slave cluster.
+
+Automated Pruning of Invalid Transactions
+-----------------------------------------
+From CDAP 4.1 onwards, CDAP supports automated pruning of the invalid transactions list.
+Since it is a beta feature, it is turned off by default. Please contact us if you are
```

Comment:
This feature is turned off by default. Please contact us at dev@cask.co if you are interested in turning it on.